Modules/ThirdParty/HDF5/src/CMakeLists.txt

```diff
@@ -64,6 +63,23 @@ set(HDF_TEST_EXPRESS OFF CACHE INTERNAL "Control testing framework (0-3)")
 set(BUILD_STATIC_EXECS OFF CACHE INTERNAL "Build Static Executables")
 set(BUILD_USER_DEFINED_LIBS OFF CACHE INTERNAL "Build user defined libs")
 
+set (HDF5_DIMENSION_SCALES_NEW_REF OFF CACHE INTERNAL "Use new-style references with dimension scale APIs" )
+set (HDF5_EXTERNAL_LIB_PREFIX "itk" CACHE INTERNAL "Use prefix for custom library naming.")
```

Comment:
@jcfr this has a potential to interact with https://github.com/Slicer/ITK/commit/aacaefd6366bb09e09faefd173d4d3797f386ad0 and other [Slicer customizations](https://github.com/Slicer/ITK/commits/slicer-v5.4.0-2024-05-16-311b706/).